ui/tests/acceptance/auth/auth-test.js

```diff
@@ -42,6 +43,14 @@ module('Acceptance | auth login form', function (hooks) {
     assert.strictEqual(currentURL(), '/vault/auth', 'invalid query param is cleared');
   });
 
+  test('it does not refire route model if query param does not exist', async function (assert) {
```

Comment:
Added this test and checked it failed before adding the query param check to `redirect`